src/views/Home/Charts/TimelineChart/index.tsx

```diff
@@ -0,0 +1,273 @@
+            const numMonthsList = timePointsWithData.map((datum) => datum.numMonths);
+
+            const xBounds = getBounds(numMonthsList);
+            const xScale = getScaleFunction(
+                xBounds,
+                { min: 0, max: bounds.width },
+                { start: chartMargin.left, end: chartMargin.right },
+            );
+
+            const numYAxisPoints = 5;
+            const approxYDiff = Math.ceil(maxValue / numYAxisPoints);
+            const newMaxValue = approxYDiff * numYAxisPoints;
+
+            const yBounds = {
+                min: 0,
+                max: newMaxValue,
+            };
+            const yScale = getScaleFunction(
+                { min: 0, max: yBounds.max },
+                { min: 0, max: bounds.height },
+                { start: chartMargin.top, end: chartMargin.bottom },
+                true,
+            );
+
+            const diff = newMaxValue / numYAxisPoints;
+            const yAxisData = Array.from(Array(numYAxisPoints + 1).keys()).map((key) => {
+                const value = yBounds.min + (diff * key);
+                return {
+                    key,
+                    dataValue: value,
+                    value: yScale(value),
+                };
+            });
+
+            const chartData = timePointsWithData.map((timePoint, i) => {
+                const x = xScale(timePoint.numMonths);
+
+                const isStart = i === 0;
+                const isEnd = i === (timePointsWithData.length - 1);
+
+                const prevDatum = isStart ? timePoint : timePointsWithData[i - 1];
+                const nextDatum = isEnd ? timePoint : timePointsWithData[i + 1];
+                const rectStart = xScale(
+                    (prevDatum.numMonths + timePoint.numMonths) / 2,
+                ) + (isStart ? -CHART_OFFSET : 0);
+                const rectEnd = xScale(
+                    (timePoint.numMonths + nextDatum.numMonths) / 2,
+                ) + (isEnd ? CHART_OFFSET : 0);
+
+                return {
+                    key: getFormattedKey(timePoint.date),
+                    x,
+                    yValues: listToMap(
+                        timePoint.dataValues,
+                        (datum) => datum.key,
+                        (datum) => yScale(datum.value),
+                    ),
+                    rect: {
+                        start: rectStart,
+                        width: Math.max(0, rectEnd - rectStart),
+                    },
+                    date: timePoint.date,
+                    numMonths: timePoint.numMonths,
+                    values: timePoint.dataValues,
+                };
+            });
+
+            return [
+                chartData,
+                yAxisData,
+            ] as const;
+        },
+        [bounds, dataKeys, timePoints, valueSelector],
+    );
+
+    const handleRectMouseClick: React.MouseEventHandler<SVGRectElement> = useCallback((e) => {
```

Comment:
We should define the type the argument `e`

Reply:
Already defined in type of callback